docs/reference/tags/variant_pricing/index.md

```diff
@@ -0,0 +1,83 @@
+---
+layout: default
+title: Variant Pricing
+parent: Tags
+grand_parent: Reference
+---
+
+# Variant Pricing
+
+The `variant_pricing` tag renders a self-updating `<span>` which can
+interactively fetch and display up-to-date pricing information for a given
+variant, provided an adult count/start date/end date combination.
+
+You can use this tag to display up-to-date prices of each variant you're
+rendering on the page, responding to user interactions with the calendar and
+adult count for each displayed variant.
+
+The tag is package-aware; if you use it in a package step template, it will
+display how much upgrading to that variant will cost, or `Included` if it is
+part of the package price.
+
+##### input
+{% raw %}
+
+    {% variant_pricing variant: variant, class: "my-nice-class", adult_count: 2, start_on: "2023-01-01", end_on: "2023-01-04" %}
+
+{% endraw %}
+
+##### output
+
+The `<span>` tag will initially be rendered empty:
+
+{% raw %}
+
+    <span class="my-nice-class"
+      data-controller="variant-pricing"
+      data-variant-pricing-variant-value="<id>"
+      data-variant-pricing-adult-count-value="2"
+      data-variant-pricing-start-on-value="2023-01-01"
+      data-variant-pricing-end-on-value="2023-01-04">
+    </span>
+
+{% endraw %}
+
+## Fetching new prices
+
+The platform provides a Javascript component which automates the triggering of
+the appropriate server requests for pricing information, interpolating the
+result into the `<span>`'s inner HTML. This component will perform the request
+at least once: when the page is fully rendered.
+
+Fetching up-to-date prices should be done in response to the following
+interactions with the page:
+
+- User selects an adult count for a variant.
+- User selects a start date for their stay.
+- User selects an end date for their stay.
+
+To support these use cases, the platform frontend component listens to changes
+the following data attributes of the `<span>` tag:
+
+- `data-variant-pricing-adult-count-value`
+- `data-variant-pricing-start-on-value`
+- `data-variant-pricing-end-on-value`
+
+We recommend listening to these changes on the page, and have the event handlers:
+
+- Target the `<span>` tags by their class 
+- Then assign the chosen values to their data-attributes 
+
+Given the `variant_pricing` tag above, one may use the following event handler:
+
+{% raw %}
+
+    function updatePrices(startOnDate, endOnDate, adultCount) {
+      const variantPricingTag = document.querySelector(".my-nice-class")
+
+      variantPricingTag.dataset.variantPricingStartOnValue = startOnDate
```

Comment:
Out of context, these variables don't mean anything. Would it be better to pass in strings with the date in the accepted format to make it clearer?